### PR TITLE
fix: deb: create subfolders and use GNU tar format

### LINKED ISSUE #6: .deb not creating directories for files to go in
I was experimenting with nfpm for rclone...

I made a simple nfpm.yaml

```yaml
name: "rclone"
arch: "amd64"
platform: "linux"
version: "1.39"
section: "default"
priority: "extra"
provides:
- rclone
maintainer: "Nick Craig-Wood <nick@craig-wood.com>"
description: |
  Rclone "rsync for cloud storage" is a command line program to sync
    files and directories to and from most cloud providers.
vendor: "rclone"
homepage: "https://rclone.org"
license: "MIT"
bindir: "/usr/bin"
files:
  ./rclone: "/usr/bin/rclone"
  ./README.html: "/usr/share/doc/rclone/README.html"
  ./README.txt: "/usr/share/doc/rclone/README.txt"
  ./rclone.1: "/usr/share/man/man1/rclone.1"
```

Which apparently works

```
$ nfpm -f ../../nfpm.yaml pkg -t rclone.deb
using deb packager...
created package: rclone.deb
```

But when I try to install the resulting .deb I get

```
$ sudo dpkg -i ./rclone.deb 
(Reading database ... 352196 files and directories currently installed.)
Preparing to unpack ./rclone.deb ...
Unpacking rclone (1.39) ...
dpkg: error processing archive ./rclone.deb (--install):
 unable to create '/usr/share/doc/rclone/README.html.dpkg-new' (while processing '/usr/share/doc/rclone/README.html'): No such file or directory
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Errors were encountered while processing:
 ./rclone.deb
```

I'm guessing that is because the directory `'/usr/share/doc/rclone/` doesn't exist and something needs to create it.

Is that something I'm supposed to do in the config file or is it a bug?

Thanks

### PR DESCRIPTION
**Pull request by caarlos0**
closes #6 

tl;dr: the problem was in the `data.tar.gz`, which had only the files, but not the folders themselves.

I had to walk through the desired path and create the headers inside the tar file, which fixes the issue as it seems.

```
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/bin/
-rwxr-xr-x  0 0      0     1034959 Feb 25 14:58 usr/bin/rclone
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/doc/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/doc/rclone/
-rw-r--r--  0 0      0        3267 Feb 25 14:58 usr/share/doc/rclone/README.txt
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/man/
drwxr-xr-x  0 0      0           0 Dec 31  1969 usr/share/man/man1/
-rw-r--r--  0 0      0      355170 Feb 25 14:58 usr/share/man/man1/rclone.1
```

before that , we had:

```
-rwxr-xr-x  0 0      0     1034959 Feb 25 14:58 usr/bin/rclone
-rw-r--r--  0 0      0        3267 Feb 25 14:58 usr/share/doc/rclone/README.txt
-rw-r--r--  0 0      0      355170 Feb 25 14:58 usr/share/man/man1/rclone.1
```

as a bonus, I also changed the format of the headers to GNU format.

--- 

TODO: 

- [x] avoid duplicating the paths (`usr/` header is added several times)
- [x] betere cover it with tests

### CODE REVIEW

**Comment by caarlos0** [comment]
> TODO: avoid duplicating the paths (usr/ header is added several times).

done

**Comment by caarlos0** [comment]
```
drwxr-xr-x  0 0      0           0 Feb 25 17:35 usr/
drwxr-xr-x  0 0      0           0 Feb 25 17:35 usr/bin/
-rwxr-xr-x  0 0      0     1034959 Feb 25 17:35 usr/bin/rclone
drwxr-xr-x  0 0      0           0 Feb 25 17:35 usr/share/
drwxr-xr-x  0 0      0           0 Feb 25 17:35 usr/share/doc/
drwxr-xr-x  0 0      0           0 Feb 25 17:35 usr/share/doc/rclone/
-rw-r--r--  0 0      0        3267 Feb 25 17:35 usr/share/doc/rclone/README.txt
```

looks great now!

**Comment by caarlos0** [comment]
```
λ docker run -v $PWD:/tmp -it ubuntu dpkg -i /tmp/rclone.deb
Selecting previously unselected package rclone.
(Reading database ... 4768 files and directories currently installed.)
Preparing to unpack /tmp/rclone.deb ...
Unpacking rclone (1.39) ...
Setting up rclone (1.39) ...
```

**Comment by caarlos0** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
